index.bs

```diff
@@ -1527,6 +1546,48 @@ example of the latter, when the user is accessing the [=[RP]=] from a given clie
 use a [=roaming authenticator=] which was originally registered with the [=[RP]=] using a different client.
 
 
+### Attestation Preference enumeration (enum <dfn enum>AttestationPreference</dfn>) ### {#attestation}
+
+[=[RPS]=] may use the {{AttestationPreference}} enum to specify their preference regarding attestation types used 
+during credential generation.
+
+<pre class="idl">
+    enum AttestationPreference {
+        "none",
+        "verifiable",
+        "direct"
+    };
+</pre>
+
+<ul>
+    <li><dfn>none</dfn> - indicates that the [=[RP]=] is not interested in attestation, and prefers fast and reliable credential
+      creation over obtaining a verifiable attestation. The client MAY replace the attestation statement generated
+      by the authenticator with a meaningless client-generated attestation statement in this case, for example in order to prevent
+      having to obtain user consent for relaying identifying information to the RP, or to save a roundtrip to a Privacy CA. 
+      The client SHOULD NOT prompt the user for consent to relay identifying information to the RP in this case, and MUST NOT 
+      fail the credential creation operation because of issues with the attestation. For example, if the client normally uses
+      a Privacy CA to anonymize attestations, and the Privacy CA is offline, the client must not fail the credential creation 
+      operation. Instead, it could replace the authenticator-generated attestation with a client-generated attestation.
+
+      This is the default value.
+    <li><dfn>verifiable</dfn> - indicates that the [=[RP]=] prefers an attestation type that leads to verifiable attestation
+      statements, but leaves it up to the client how to obtain such attestation statements. The client MAY replace the
+      authenticator-generated attestation statements with attestation statements generated by a Privacy CA, in order to protect
+      the privacy of the user, or to assist [=RPS=] with attestation verification in a heterogeneous ecosystem.
+      
+      Note that there is no guarantee that the [=RP=] will obtain a verifiable attestation statement in this case. The authenticator
+      may use self-attestation, or the user may have opted out of relaying any kind of (identifying) attestation information to
+      the RP
+    <li><dfn>direct</dfn> - indicates that the [=[RP]=] wants the attestation statement as generated by the authenticator. If
+      the client returns an attestation statement to the RP in this case, it MUST be the statement generated by the authenticator.
+      The client SHOULD obtain consent from the user in this case for relaying attestation information to the [=RP=].
```

Comment:
Why should platform penalize RP by adding user friction who ask for it legitimately and authenticators who are following all the rules set out for batch attestation?

I can understand consent situations when in incognito mode, but don't understand the normal case here.

Reply:
The "obtain user consent" handling should be identical for "verifiable" and "direct". In both cases the authenticator model will be known to some other party (either the Privacy CA or the RP).
In my opinion Authenticator models satisfying the FIDO requirements (i.e. batch size >= 100000 and others) do not lead to a privacy risk and hence do not require yet another user consent (in addition to the general consent of using the authenticator a creating a credential).